packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/snapshot_control_pane.dart

```diff
@@ -36,17 +37,39 @@ class SnapshotControlPane extends StatelessWidget {
                       current: current,
                       controller: controller,
                     ),
-                    const SizedBox(width: defaultSpacing),
+                    const SizedBox(width: denseSpacing),
                     SizedBox(
                       width: _classFilterWidth,
                       child: _ClassFilter(onChanged: controller.setClassFilter),
                     ),
-                    const SizedBox(width: defaultSpacing),
+                    const SizedBox(width: denseSpacing),
                     ToCsvButton(
                       minScreenWidthForTextBeforeScaling:
                           primaryControlsMinVerboseWidth,
                       onPressed: current.downloadToCsv,
                     ),
+                    const SizedBox(width: denseSpacing),
+                    ValueListenableBuilder<ClassStats?>(
+                      valueListenable: controller.derived.selectedClassStats,
+                      builder: (_, theClass, ___) => CopyButton(
+                        tooltip: 'Copy full name of the selected class.',
+                        contentBuilder: theClass == null
+                            ? null
+                            : () => theClass.heapClass.fullName,
+                        label: 'Class',
+                      ),
+                    ),
+                    const SizedBox(width: denseSpacing),
+                    ValueListenableBuilder<StatsByPathEntry?>(
+                      valueListenable: controller.derived.selectedPathEntry,
+                      builder: (_, pathEntry, ___) => CopyButton(
+                        tooltip: 'Copy the selected retaining path.',
+                        contentBuilder: pathEntry == null
+                            ? null
+                            : () => pathEntry.key.asLongString(),
+                        label: 'Path',
+                      ),
+                    ),
```

Comment:
instead of adding these copy controls to the button bar, I think we should mimc what is done on the network page. This will help with consistency and will make the action bar less busy.
<img width="826" alt="Screen Shot 2022-10-10 at 9 29 26 PM" src="https://user-images.githubusercontent.com/43759233/194977294-fa7e2a3a-e681-4f4b-89ea-582b011587bc.png">
<img width="227" alt="Screen Shot 2022-10-10 at 9 29 31 PM" src="https://user-images.githubusercontent.com/43759233/194977299-ddb61e7e-ff6b-4eb2-82fc-60a89b90b93c.png">